### ISSUE TITLE
Gitlab pipelines might mark as "broken spec" build that failed because of system failures

### ISSUE THREAD

**Issue by alalazo** [spack/spack-infrastructure#324]
Inspecting the [list of pipelines run on develop](https://gitlab.spack.io/spack/spack/-/pipelines?page=2&scope=all&ref=develop) there is a system failure when building `mpich` occurring here:
- https://gitlab.spack.io/spack/spack/-/pipelines/173449/failures

The pipeline running after the one above fails, reporting the two `mpich` as broken specs:
- https://gitlab.spack.io/spack/spack/-/jobs/3334734

https://github.com/spack/spack/pull/32839 swaps two lines of code to trigger a rebuild of `mpich`, which results in the following jobs for the `aws-isc` pipeline:
- https://gitlab.spack.io/spack/spack/-/pipelines/176945

where `mpich` builds fine. That seems to suggest that a system failure was recorded as a broken spec.

**Comment by scottwittenburg** [comment]
Just pointing out that if the rebuild job can never get scheduled onto a pod, as in the jobs you linked above, then we definitely don't report anything to the broken specs list, as none of our rebuild script logic ever gets to run.  In the case of the broken `mpich/oxqu5m7`, it appears to have been recorded on the broken specs list here: https://gitlab.spack.io/spack/spack/-/jobs/3333864.  Then the other retries were already used up, or had previously failed and this was the third one.

**Comment by alalazo** [comment]
Feel free to change the title of the spec to make it more precise. What I did to track this down was:
1. Get a list of the pipelines on `develop`
2. Look one by one at the failed pipelines to see when the failure was happening
3. Check "failed jobs" as in the screenshot

![Screenshot from 2022-09-27 19-19-54](https://user-images.githubusercontent.com/4199709/192593699-4ac4ac94-fc26-407c-aa28-cf5decc48a7c.png)

Do I understand correctly then that "Failed jobs" might not be the job that really failed, but I need to look further down in the right column, like:

![column_right](https://user-images.githubusercontent.com/4199709/192594476-71761eeb-1316-43d4-afaf-4ebed34dc0ef.png)

**Comment by scottwittenburg** [comment]
Yes, by looking down that list at the other failures of the same hash, I found the one that actually got scheduled, ran, and the build failed (and said it was going to report the breakage).
